src/node/mod.rs

```diff
@@ -528,12 +521,12 @@ impl<N: NodeData<J>, J: Jet> Node<N, J> {
     /// See the documentation for [`Converter`] for details.
     pub fn convert<S, M, C>(&self, converter: &mut C) -> Result<Arc<Node<M, J>>, C::Error>
     where
-        S: for<'a> SharingTracker<SwapChildren<&'a Self>> + Default,
+        S: for<'a> SharingTracker<&'a Self> + Default,
         M: NodeData<J>,
         C: Converter<N, M, J>,
     {
         let mut converted: Vec<Arc<Node<M, J>>> = vec![];
-        for data in self.rtl_post_order_iter::<S>() {
+        for data in self.post_order_iter::<S>() {
```

Comment:
e97fbc8f3ffb9eb3aaf9f35dca953a4b9e64048b: Why does LTR / RTL not affect any unit tests? Is it because children are still visited before parents, and no implementation of `Converter` tracks the nodes it has seen? As long as `convert_witness()` and `convert_data()` are called with the right arguments (which they are) the same DAG will be built.

Reply:
Yep, exactly. There's no "local" way to tell what direction the iterator is going in, so you would need to somehow be tracking nodes you've seen before...and the `convert` function is designed so that you don't need to do that.

The only place it would've mattered was in my old finalization code, which had some insane logic where it would be iterating an AST in parallel to the Simplicity DAG, and caching random bits of data which only made sense assuming a particular iteration order. I think it's good for multiple reasons that we decided to throw that away :)